test/src/responder/multiResponder.test.ts

```diff
@@ -0,0 +1,277 @@
+        const appointmentId = "app1";
+        const responseData = createResponseData("app1");
+
+        const responder = new MultiResponder(
+            signer,
+            increasingGasPriceEstimator,
+            transactionTracker,
+            maxConcurrentResponses,
+            replacementRate
+        );
+
+        await responder.startResponse(appointmentId, responseData);
+
+        verify(transactionTrackerMock.addTx(anything(), anything())).once();
+    });
+
+    it("startResponse can issue two transactions and replace", async () => {
+        const appointmentId = "app1";
+        const responseData = createResponseData("app1");
+        const appointmentId2 = "app2";
+        const responseData2 = createResponseData("app2");
+
+        const responder = new MultiResponder(
+            signer,
+            increasingGasPriceEstimator,
+            transactionTracker,
+            maxConcurrentResponses,
+            replacementRate
+        );
+
+        await responder.startResponse(appointmentId, responseData);
+        verify(transactionTrackerMock.addTx(anything(), anything())).once();
+        // because the gas price is increasing this should result in a replacement
+        // therefor two additional transactions are issued, rather than just one
+        await responder.startResponse(appointmentId2, responseData2);
+        verify(transactionTrackerMock.addTx(anything(), anything())).times(3);
+    });
+
+    it("startResponse can issue two transactions but not replace", async () => {
+        const appointmentId = "app1";
+        const responseData = createResponseData("app1");
+        const appointmentId2 = "app2";
+        const responseData2 = createResponseData("app2");
+
+        const responder = new MultiResponder(
+            signer,
+            decreasingGasPriceEstimator,
+            transactionTracker,
+            maxConcurrentResponses,
+            replacementRate
+        );
+
+        await responder.startResponse(appointmentId, responseData);
+        verify(transactionTrackerMock.addTx(anything(), anything())).once();
+        // because the gas price is decreasing this should result not result in a replacement
+        // therefore only one new transaction should be issued
+        await responder.startResponse(appointmentId2, responseData2);
+        verify(transactionTrackerMock.addTx(anything(), anything())).times(2);
+    });
+
+    it("startResponse swallows error", async () => {
+        const appointmentId = "app1";
+        const responseData = createResponseData("app1");
+
+        const responder = new MultiResponder(
+            signer,
+            errorGasPriceEstimator,
+            transactionTracker,
+            maxConcurrentResponses,
+            replacementRate
+        );
+
+        await responder.startResponse(appointmentId, responseData);
+
+        verify(transactionTrackerMock.addTx(anything(), anything())).never();
+    });
```

Comment:
How does the caller of `startResponse` know that the response was not started successfully? Who is responsible for retrying?
For the dedicated responder, events are emitted and the ResponderManager could react accordingly; what is the plan here?

Reply:
The plan in the long run will be to throw exceptions and handle these at the responder manager level, where we can retry or ask call other responders. For now I dont want errors to propogate to the watcher, so I but a large try/catch block here, but I could put one in the responder manager instead if you think that's better? My long term plan is to get this working in the anchor state paradigm, and then re-assess how these classes will interact.

Reply:
Fair enough, throwing errors in the Responders and handling them at the level of the ResponderManager seems appropriate to me.